ouroboros-consensus-shelley/test/Test/ThreadNet/RealTPraos.hs

```diff
@@ -228,7 +232,7 @@ mkGenesisConfig k maxKESEvolutions coreNodes = ShelleyGenesis {
     , sgNetworkMagic          = NetworkMagic 0
     , sgProtocolMagicId       = ProtocolMagicId 0
     , sgActiveSlotsCoeff      = 0.5 -- TODO 1 is not accepted by 'mkActiveSlotCoeff'
-    , sgDecentralisationParam = 1
+    , sgDecentralisationParam = 0
```

Comment:
**UPDATE:** I see you vary `d` it in a later commit.

---

Shouldn't we test with different values for this parameter? We can make it an input parameter and generate it. Hmm, I see now that it's not yet possible to a have block/protocol-specific arbitrary parameter in the `TestConfig`. There are some options:

1. We introduce a new data type containing `SecurityParam`, `TestConfig`, and any `RealTPraos`-specific types, such as this `d`. In the `Arbitrary` instance for it, we can use the existing generators and shrinkers. This helps us get rid of the `forAll` and `forAllShrink` in the `testProperty`.
2. We use another `forAll(Shrink` for `d` in the `testProperty`. Note that such nested `forAll`s don't allow proper shrinking, as @edsko would say.
3. We use `MonadRandom` (e.g., `generateBetween` with some scaling) and `withSeed` to generate such a `d` in `prop_simple_real_tpraos_convergence`, just like we do for the core nodes.